ovsdb/client.go

```diff
@@ -229,6 +277,45 @@ func (c *Client) listen() {
 	}
 }
 
+// echoLoop starts a loop that sends echo RPCs at the interval defined by d.
+func (c *Client) echoLoop(ctx context.Context, d time.Duration) {
+	t := time.NewTicker(d)
+	defer t.Stop()
+
+	for {
+		// Prioritize closed context over a tick on the channel.
+		select {
+		case <-ctx.Done():
```

Comment:
Could you get rid of the nested select by doing something like this?

```go
select {
    case <-ctx.Done():
        return
    case <-t.C:
        if err := ctx.Err(); err != nil {
            return
        }
}
```

Since `Err()` only returns non-nil when `Done()` is closed -- if both channels are ready but it takes the ticker path, this gives you way to check for a closed context there.

Reply:
I like this and will implement it if we end up keeping this PR.